kudu-connector/src/main/java/com/sequenceiq/cloudbreak/kudu/processor/AbstractKuduRecordProcessor.java

```diff
@@ -0,0 +1,24 @@
+package com.sequenceiq.cloudbreak.kudu.processor;
+
+import java.util.concurrent.BlockingDeque;
+
+import com.sequenceiq.cloudbreak.kudu.config.KuduConfiguration;
+import com.sequenceiq.cloudbreak.kudu.model.KuduRecordRequest;
+import com.sequenceiq.cloudbreak.streaming.processor.AbstractRecordProcessor;
+
+import io.opentracing.Tracer;
+
+public abstract class AbstractKuduRecordProcessor extends AbstractRecordProcessor<KuduConfiguration, KuduRecordRequest, KuduRecordWorker> {
+
+    private final Tracer tracer;
+
+    public AbstractKuduRecordProcessor(KuduConfiguration configuration, Tracer tracer) {
+        super(configuration);
+        this.tracer = tracer;
+    }
+
+    @Override
+    public KuduRecordWorker createWorker(String threadName, BlockingDeque<KuduRecordRequest> processingQueue) {
+        return new KuduRecordWorker(threadName, getServiceName(), this, processingQueue, getConfiguration(), tracer);
+    }
```

Comment:
I'd also override the config validation function here like in the case of the Kafka implementation